biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -112,6 +135,12 @@ public File apply(File t) {
 		return link;
 	}
 
+	private boolean needsCopy(Path sourcePath, Path destPath) {
+		final File dest = destPath.toFile();
+
+		return !dest.exists() || dest.length() != sourcePath.toFile().length();
```

Comment:
I agree we need more than content length, we need sha or md5 to truly determine if we need to recopy.  Because there are bad behaving p2 repos out there, that keep the version and bsn the same, but might change the content and the content length accidentally stays the same (even though the chances here are pretty remote).